list.jsx

```diff
@@ -0,0 +1,83 @@
+import React, { PureComponent, createRef } from 'react'; 
```

Comment:
import React, { useEffect, useRef, useState } from "react";

type DataListProps = {
  height: string;
  parentRef: Element;
  onItemClick: () => void;
  sendCoordinates: () => void;
};

type DataItem = {
  id: string;
  value: string;
};

export const DataList = ({ height, parentRef }: DataListProps) => {
  const [listHeight, setListHeight] = useState(height);
  const { data, isLoading, error } = useData();
  const ref = useRef<HTMLElement>(null);

  const handleParentContainerScroll = () => {
    if (ref) {
      const rect = ref.getBoundingClientRect();

      this.props.sendCoordinates(rect);
    }
  };

  useEffect(() => {
    parentRef.addEventListener("scroll", this.handleParentContainerScroll);
  }, []);

  if (error) return <div>Error occured</div>;

  return <div style={{}} ref={ref}></div>;
};